File: rust/src/commands/vaults.rs
use clap::Subcommand;
use colored::*;

use crate::vault;

#[derive(Subcommand, Clone)]
pub enum VaultCommands {
    /// List known vault repositories
    List,
    /// Show the currently selected vault repository
    Current,
    /// Interactively choose the active vault repository
    Select,
}

pub fn run(command: VaultCommands) {
    match command {
        VaultCommands::List => list_vaults(),
        VaultCommands::Current => current_vault(),
        VaultCommands::Select => select_vault(),
    }
}

fn list_vaults() {
    let current = vault::current_vault_dir();
    let vaults = vault::list_known_vaults();

    if vaults.is_empty() {
        println!(
            "  {} No known vaults found under {}",
            "!".yellow().bold(),
            vault::vaults_home_dir().display()
        );
        return;
    }

    println!("\n  {}\n", "Known Vaults:".bold().underline());
    for path in vaults {
        let marker = if current.as_ref() == Some(&path) { "*" } else { "-" };
        println!("  {} {}", marker.cyan().bold(), path.display());
    }
    println!();
}

fn current_vault() {
    match vault::current_vault_dir() {
        Some(path) => println!("{}", path.display()),
        None => {
            eprintln!(
                "No active vault is selected. Run `keypick vault select` or `keypick setup`."
            );
            crate::terminal::cleanup_and_exit(1);
        }
    }
}

fn select_vault() {
    match vault::select_known_vault_interactively() {
        Ok(path) => {
            println!(
                "\n  {} {}",
                "Active vault:".green().bold(),
                path.display().to_string().cyan().bold()
            );
        }
        Err(error) => {
            eprintln!("{}", error);
            crate::terminal::cleanup_and_exit(1);
        }
    }
}

File: ts/src/commands/vaults.ts
// Vault repository management subcommands.

import chalk from "chalk";
import * as vault from "../lib/vault.ts";
import * as terminal from "../lib/terminal.ts";

export type VaultSubcommand = "list" | "current" | "select";

export async function run(sub: VaultSubcommand): Promise<void> {
  switch (sub) {
    case "list":
      return listVaults();
    case "current":
      return currentVault();
    case "select":
      return selectVault();
  }
}

function listVaults(): void {
  const current = vault.currentVaultDir();
  const vaults = vault.listKnownVaults();

  if (vaults.length === 0) {
    console.log(
      `  ${chalk.yellow.bold("!")} No known vaults found under ${vault.vaultsHomeDir()}`,
    );
    return;
  }

  console.log(`\n  ${chalk.bold.underline("Known Vaults:")}\n`);
  for (const p of vaults) {
    const marker = current === p ? "*" : "-";
    console.log(`  ${chalk.cyan.bold(marker)} ${p}`);
  }
  console.log();
}

function currentVault(): void {
  const p = vault.currentVaultDir();
  if (p) {
    console.log(p);
  } else {
    console.error("No active vault is selected. Run `keypick vault select` or `keypick setup`.");
    terminal.cleanupAndExit(1);
  }
}

async function selectVault(): Promise<void> {
  try {
    const p = await vault.selectKnownVaultInteractively();
    console.log(`\n  ${chalk.green.bold("Active vault:")} ${chalk.cyan.bold(p)}`);
  } catch (e) {
    console.error((e as Error).message);
    terminal.cleanupAndExit(1);
  }
}
